test/test_e2e.rb

```diff
@@ -38,6 +38,14 @@ def test_flakey_suite
     assert_equal({ "./spec/foo_spec.rb[1:1]" => "2" }, queue.requeued_jobs)
   end
 
+  def test_suite_with_failures_without_retries
+    queue = exec_build("failing_suite", "--max-requeues=0")
```

Comment:
Since that suite would fail anyway (even with retries), we should really use the same suite used in the `test_flakey_suite`. Actually we could copy the body of that test and also assert on `assert_processed_jobs`, for completeness.